hugegraph-api/src/main/java/com/baidu/hugegraph/api/graph/UpdateStrategy.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2017 HugeGraph Authors
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.baidu.hugegraph.api.graph;
+
+import java.math.BigDecimal;
+import java.util.Collection;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import com.baidu.hugegraph.util.E;
+import com.baidu.hugegraph.util.NumericUtil;
+import com.google.common.collect.Sets;
+
+public enum UpdateStrategy {
+
+    // Only number support sum
+    SUM {
+        @Override
+        Object updatePropertyValue(Object oldProperty, Object newProperty) {
+            // TODO: Improve preformance? (like write a method in common module)
+            BigDecimal oldNumber = new BigDecimal(oldProperty.toString());
+            BigDecimal newNumber = new BigDecimal(newProperty.toString());
+            return oldNumber.add(newNumber);
+        }
+
+        @Override
+        void checkPropertyType(Object oldProperty, Object newProperty) {
+            E.checkArgument(oldProperty instanceof Number &&
+                            newProperty instanceof Number,
+                            formatError(oldProperty, newProperty, "Number"));
+        }
+    },
+
+    // Only Date & Number support compare
+    BIGGER {
+        @Override
+        Object updatePropertyValue(Object oldProperty, Object newProperty) {
+            return compareNumber(oldProperty, newProperty, BIGGER);
+        }
+
+        @Override
+        void checkPropertyType(Object oldProperty, Object newProperty) {
+            E.checkArgument((oldProperty instanceof Date ||
+                             oldProperty instanceof Number) &&
+                            (newProperty instanceof Date ||
+                             newProperty instanceof Number),
```

Comment:
(oldProperty instanceof Date && newProperty instanceof Date) ||
(oldProperty instanceof Number && newProperty instanceof Number)

Reply:
There are at least 2 situation:
1. one is `Date` type, and the other is a `Number` type.  
2. both are `Number` type.

the `list/set` collection has the same reason.

Reply:
ok